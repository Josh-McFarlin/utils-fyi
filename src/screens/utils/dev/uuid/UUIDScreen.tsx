import * as React from "react";
import clsx from "clsx";
import { v4 as uuidv4 } from "uuid";
import { saveAs } from "file-saver";
import Button from "../../../../components/Button";
import FormInput from "../../../../components/FormInput";
import MultilineTextOutput from "../../../../components/MultilineTextOutput";
import classes from "./UUIDScreen.module.scss";

const UUIDScreen = () => {
  const [uuid, setUUID] = React.useState<string>(uuidv4());
  const [showList, setShowList] = React.useState<boolean>(false);
  const [listCount, setCount] = React.useState<number>(5);
  const [uuidList, setList] = React.useState<string[]>([]);

  const createUUID = () => uuidv4();

  const generateList = () => {
    const newList = Array.from({ length: listCount }, () => createUUID());

    setList(newList);
    setShowList(true);
  };

  const uuidListText =
    uuidList != null && uuidList.length > 0 ? uuidList.join("\n") : "";

  const handleDownload = () => {
    const blob = new Blob([uuidListText], {
      type: "text/plain;charset=utf-8",
    });

    saveAs(blob, "uuids (utils-fyi).txt");
  };

  return (
    <>
      <h1>UUID v4</h1>
      <div className={clsx(classes.row, classes.spaceBottom)}>
        <h2 className={classes.spaceRight}>UUID: {uuid}</h2>
        <Button onClick={() => setUUID(createUUID)}>Generate New</Button>
      </div>

      <h3>Generate List</h3>
      <div className={clsx(classes.row, classes.spaceBottom)}>
        <FormInput
          className={classes.spaceRight}
          label="Count"
          type="number"
          min={1}
          max={500}
          value={listCount}
          onChange={(event) => setCount(parseInt(event.target.value, 10))}
        />
        <Button onClick={generateList}>Generate List</Button>
      </div>
      {showList && (
        <>
          <MultilineTextOutput>{uuidListText}</MultilineTextOutput>
          <div className={clsx(classes.row, classes.spaceTop)}>
            <Button className={classes.spaceRight} onClick={handleDownload}>
              Download List
            </Button>
            <Button variant="warning" onClick={() => setShowList(false)}>
              Hide
            </Button>
          </div>
        </>
      )}
    </>
  );
};

export default UUIDScreen;
