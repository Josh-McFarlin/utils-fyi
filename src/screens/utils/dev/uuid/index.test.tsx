import React from "react";
import { render } from "../../../../../test/testUtils";
import UUIDScreen from "./UUIDScreen";

jest.mock("uuid", () => ({
  v4: () => "00000000-0000-0000-0000-000000000000",
}));

describe("UUID Page", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<UUIDScreen />, {});

    expect(asFragment()).toMatchSnapshot();
  });
});
