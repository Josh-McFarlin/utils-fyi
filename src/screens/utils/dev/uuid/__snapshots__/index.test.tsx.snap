// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UUID Page matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    UUID v4
  </h1>
  <div
    class="row spaceBottom"
  >
    <h2
      class="spaceRight"
    >
      UUID: 00000000-0000-0000-0000-000000000000
    </h2>
    <button
      class="root primaryColor"
      type="button"
    >
      Generate New
    </button>
  </div>
  <h3>
    Generate List
  </h3>
  <div
    class="row spaceBottom"
  >
    <label
      class="root spaceRight"
    >
      <p>
        Count
      </p>
      <input
        max="500"
        min="1"
        type="number"
        value="5"
      />
    </label>
    <button
      class="root primaryColor"
      type="button"
    >
      Generate List
    </button>
  </div>
</DocumentFragment>
`;
